src/snowflake/cli/_plugins/nativeapp/sf_sql_facade.py

```diff
@@ -136,6 +155,31 @@ def _use_schema_optional(self, schema_name: str | None):
         """
         return self._use_object_optional(UseObjectType.SCHEMA, schema_name)
 
+    def _grant_privileges_to_role(
+        self,
+        privileges: list[str],
+        object_type: ObjectType,
+        object_identifier: str,
+        role_to_grant: str,
+        role_to_use: str | None = None,
+    ) -> None:
+        """
+        Grants one or more access privileges on a securable object to a role
+
+        @param privileges: List of privileges to grant to a role
+        @param object_type: Type of snowflake object to grant to a role
+        @param object_identifier: Valid identifier of the snowflake object to grant to a role
+        @param role_to_grant: Name of the role to grant privileges to
+        @param [Optional] role_to_use: Name of the role to use to grant privileges
+        """
+        comma_separated_privileges = ", ".join(privileges)
+        object_type_and_name = f"{object_type.value.sf_name} {object_identifier}"
+
+        with self._use_role_optional(role_to_use):
+            self._sql_executor.execute_query(
```

Comment:
No error handling on this to defer to the caller to categorize the errors (e.g. whether or not we can safely say it's user vs. our errors), let me know any thoughts

Reply:
what if we just plug in the generic_error_handling function here?